Makefile

```diff
@@ -1,27 +1,92 @@
-TOOLCHAIN_PATH=/opt/riscv64-unknown-elf-gcc-8.3.0-2020.04.1-x86_64-linux-ubuntu14/bin/
-FLASH_PATH=../flasher/ft2232h
-TOOLCHAIN_PREFIX=riscv64-unknown-elf
+#
+# Configure your RiscV toolchain path and prefix
+#
+TOOLCHAIN_PATH		?= /opt/riscv64-unknown-elf-gcc-8.3.0-2020.04.1-x86_64-linux-ubuntu14/bin/
+TOOLCHAIN_PREFIX	?= riscv64-unknown-elf-
 
-flash_tigard: ${PATTERN}.bin
-	PYTHONPATH=$(FLASH_PATH) python3 $(FLASH_PATH)/caravelflash/flash_util.py  --write $<
+# This path appears to be outside this project tree, so maybe instructions here to download this tool ?
+FLASH_PATH		?= ../flasher/ft2232h
 
-%.elf: %.c ../deps/sections.lds ../deps/crt0_vex.S
-	$(TOOLCHAIN_PATH)$(TOOLCHAIN_PREFIX)-gcc -I../deps -I../deps/generated/ -O0 -mabi=ilp32 -march=rv32i -D__vexriscv__ -Wl,-Bstatic,-T,../deps/sections.lds,--strip-debug -ffreestanding -nostdlib -o $@ ../deps/crt0_vex.S ../deps/isr.c ../deps/stub.c $<
-	${TOOLCHAIN_PATH}$(TOOLCHAIN_PREFIX)-objdump -D $@ > ${PATTERN}.lst
+
+ifeq ($(MAKELEVEL),0)
+ ifeq (deps,$(wildcard deps))
```

Comment:
this is so the makefile behaves differently if it's included?

Reply:
Any toplevel `make` process has MAKELEVEL=0 this may be executed from the toplevel directory, or from the sub-directory.  The outer make process that is created by the interactive user running the command from a shell runs with MAKELEVEL=0.

Inside the Makefile there is `$(MAKE) -C subdir target` this is requesting make to run make subprocess, when this occurs, a sub-process is created, the current directory is changed due to -C `subdir` and the `make target` command is executed from there.  This subprocess has a MAKELEVEL=1 context, because it is a child process of a superior make process.

This is used to detect is we are the toplevel MAKE process or a sub-ordinate make process.  So only a toplevel make process gets to setup/initialize and export or not-export some variables.